pennylane/ops/op_math/prod.py

```diff
@@ -300,7 +300,14 @@ def matrix(self, wire_order=None):
         batched: List[bool] = []  # batched[i] tells if mats[i] is batched or not
         for ops in self.overlapping_ops:
             gen = (
-                (qml.matrix(op) if isinstance(op, Hamiltonian) else op.matrix(), op.wires)
+                (
+                    (
+                        qml.matrix(op)
+                        if isinstance(op, (qml.ops.Hamiltonian, LinearCombination))
```

Comment:
```suggestion
                        if isinstance(op, qml.ops.Hamiltonian))
```
`LinearCombaintion` should define it's matrix.